src/middleware.ts

```diff
@@ -81,6 +81,11 @@ export const devtools = <S extends State>(
         } else {
           savedSetState(JSON.parse(message.state))
         }
+      } else if (
+        message.type === 'DISPATCH' &&
+        message.payload?.type === 'COMMIT'
+      ) {
+        api.devtools.init(api.getState())
```

Comment:
@dai-shi I am not really familiar with the code above, especially what `ignoreState` is doing, since `setState` is called even if `ignoreState` is true ... 🤔 

maybe an explicit switch on the `message.payload.type` like we are seeing [here](https://github.com/aurelia/store/commit/86b7edab8efc025c7fa10491db66bd305f338922#diff-cfb8e44b6757fd740446e9e4486b73c1R337) might be better?

Reply:
@TkDodo I'm not so sure either. You can clean them up as long as you confirm working correctly.

Be careful with `if (!api.dispatch` though, it's to work with the `redux` middleware. You might need to test with it too, if you haven't.

Reply:
Okay, in that case, I‘d rather leave it as-is 😅

Reply:
In any case, we'd need to check if this PR works with `redux` middleware.

Reply:
here is an example of this PR with the redux middleware - works nicely for me, including the commit button: https://codesandbox.io/s/admiring-ardinghelli-28k1c

Reply:
Thanks!! Confirmed. Working great.